### PR TITLE
ITSU-62: Initial commit of rsync backup role

### PR DESCRIPTION
**Pull request by SlothOfAnarchy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# rsync-backup
+
+This role deploys a Systemd service that backes up a filesystem using rsync.
```
Comment (dasJ) [inline README.md]: This is not about [Backes](https://de.wikipedia.org/wiki/Wieland_Backes). Also Systemd is illegal. Also you should explain that it has the capability to perform remote backups.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Requirements
+
+Debian.
```
Comment (dasJ) [inline README.md]: Cool & Good

Comment (dasJ) [reply README.md]: -.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+| `rsync_backup_target_path` | `/mnt/backup`                                | Destination path to backup to.                                                                                   |
+| `rsync_backup_mount_pre`   | `True`                                       | Whether to mount the target path before rsyncing (implies that there has to be a entry for it in `/etc/fstab`).  |
+| `rsync_backup_umount_post` | `True`                                       | Whether to unmount the target path after rsyncing (implies that there has to be a entry for it in `/etc/fstab`). |
+| `rsync_backup_flags`       | _see [defaults/main.yml](defaults/main.yml)_ | A list of flags to set when executing rsync, like `delete-during` or `stats`.                                    |
+| `rsync_backup_short_flags` | _see [defaults/main.yml](defaults/main.yml)_ | A list of single-char flags to set when executing rsync, like `a` or `h`.                                        |
+| `rsync_backup_excludes`    | _see [defaults/main.yml](defaults/main.yml)_ | A list of paths to exclude from backing up.                                                                      |
```
Comment (dasJ) [inline README.md]: Please elaborate that paths have to be relative

File: README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+| `rsync_backup_frequency`   | `daily`                                      | Systemd execution frequency of the timer.                                                                        |
```
Comment (dasJ) [inline README.md]: *systemd

File: README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## Example Playbook
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
Comment (dasJ) [inline README.md]: No.

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    mode: 0644
+    src: rsync-backup.timer.j2
+  register: rsynctimer
+
+- name: Enable and start systemd timer
+  systemd:
+    name: rsync-backup.timer
+    enabled: True
+    state: started
+    daemon_reload: "{{ rsyncservice.changed or rsynctimer.changed }}"
```
Comment (dasJ) [inline tasks/main.yml:29]: This is deprecated

File: templates/rsync-backup.service.j2 [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+Description=rsync backup from {{ rsync_backup_source_path }} to {{ rsync_backup_target_path }}
+
+[Service]
+{% if rsync_backup_mount_pre -%}
+	ExecStartPre=/bin/sh -c '/bin/mkdir -p "{{ rsync_backup_target_path }}" && /bin/mount "{{ rsync_backup_target_path }}'
+{%- endif %}
+
+ExecStart=/bin/sh -c '/usr/bin/rsync
+{%- if rsync_backup_short_flags | length > 0 %}{{ " -" }}
+	{%- for flag in rsync_backup_short_flags -%}
```
Comment (dasJ) [inline templates/rsync-backup.service.j2]: Please implode that

**Review by dasJ — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 6 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml:29]: No.
```yaml
daemon_reload: rsyncservice.changed or rsynctimer.changed
```
> rsyncservice.changed or rsynctimer.changed is not a valid boolean.  Valid booleans include: 0, 'on', 'f', 'false', 1, 'no', 'n', '1', '0', 't', 'y', 'off', 'yes', 'true'

**Comment by SlothOfAnarchy** [comment]
please rereview

**Review by dasJ — APPROVED** [review]

**Pull request merged**